auth/grants.go

```diff
@@ -19,16 +19,44 @@ import (
 	"strings"
 
 	"golang.org/x/exp/slices"
+	"google.golang.org/protobuf/encoding/protojson"
 
 	"github.com/livekit/protocol/livekit"
+	"github.com/livekit/protocol/utils"
 )
 
+type RoomConfiguration livekit.RoomConfiguration
+
+var tokenMarshaler = protojson.MarshalOptions{
+	EmitDefaultValues: false,
+}
+
+func (c *RoomConfiguration) Clone() *RoomConfiguration {
+	if c == nil {
+		return nil
+	}
+	return (*RoomConfiguration)(utils.CloneProto((*livekit.RoomConfiguration)(c)))
+}
+
+func (c *RoomConfiguration) MarshalJSON() ([]byte, error) {
+	return tokenMarshaler.Marshal((*livekit.RoomConfiguration)(c))
+}
+
+func (c *RoomConfiguration) UnmarshalJSON(data []byte) error {
+	return protojson.Unmarshal(data, (*livekit.RoomConfiguration)(c))
+}
+
 type ClaimGrants struct {
 	Identity string      `json:"-"`
 	Name     string      `json:"name,omitempty"`
 	Kind     string      `json:"kind,omitempty"`
 	Video    *VideoGrant `json:"video,omitempty"`
 	SIP      *SIPGrant   `json:"sip,omitempty"`
+	// Room configuration to use if this participant initiates the room
+	RoomConfig *RoomConfiguration `json:"roomConfig,omitempty"`
+	// Cloud-only, config preset to use
```

Comment:
in oss this is loaded from `RoomConfig.RoomConfigurations`

Reply:
yeah. we could change that as well, to be a bit more concise.

Reply:
These can be quite large. Are we ok with long tokens?

Reply:
yeah I think that's fine.. if their use case requires it.

for egress usage though, we'll need to advise users to not put their egress keys in there, but instead to rely on "default storage configurations" that's specific to the project.